### PR TITLE
Naive search for titles

### PR DESCRIPTION
**Pull request by felipesere**
Do a very basic search against both repos looking for matching titles.

Linking out to the search results is done in a special method in `search/view.ex` to cater for the different types.

### CODE REVIEW

File: test/support/sample_data.ex [hunk 1]
```diff
@@ -0,0 +1,32 @@
+defmodule Lighthouse.SampleData do
+  alias Lighthouse.Books.Book
+  alias Lighthouse.Papers.Paper
+
+  def sample_paper do
+    sample_paper("My Fancy Paper")
+  end
+
+  def sample_book() do
```
Comment (marosluuce) [inline test/support/sample_data.ex]: `()` intentional?

**Comment by marosluuce** [comment]
I almost feel that we should merge this as is. I'll start another branch to work out the ecmascript6 + testing issues.

**Comment by Maikon** [comment]
Squash hacky commits :zap:

**Comment by heruku** [comment]
:+1: to Makis, squash these commits and :shipit:

File: web/static/js/search-handler.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+(function () {
+   window.SearchHandler = (function() {
+     return  {
+       bindEvents : function(target) { 
+         var targetSelector = "[data-id='"+target+"']";
```
Comment (heruku) [inline web/static/js/search-handler.js]: I'm not sure about this concatenation. Passing in the whole `data-id="search"` _or_ just hardcoding it on this line feels cleaner.

Comment (heruku) [inline web/static/js/search-handler.js]: Another point on this. I don't know enough to make a strong case either way, but should we use `data-id="foo"` or just `data-foo`? Or not even prefix with `data` at all? In theory you can just slap a custom attribute to the search field like this: `<input lighthouse-search type="text">` and find it with jquery : `$("[lighthouse-search]")` .

Comment (felipesere) [inline web/static/js/search-handler.js]: I tried changing it in the spec and it failed with just `<input lighthouse-search ...>`.
Other than that: I don't know which commits you think I should squash ;)

**Pull request merged**